content/en/docs/reference/labels-annotations-taints.md

```diff
@@ -36,7 +36,13 @@ Example: `kubernetes.io/metadata.name=mynamespace`
 
 Used on: Namespaces
 
-Kubernetes API Server defaults this label to the namespace name during admission. This label can be used with any namespace selector, as an example with NetworkPolicy objects.
+When the `NamespaceDefaultLabelName`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled,
+the Kubernetes API server defaults newly-added namespaces to have this label set to the
+name of the namespace.
```

Comment:
If I'm understanding correctly, an attacker with access to write labels to namespaces could overwrite the default label. Have I got that right?

(Also, I'm not sure what the behavior is for clusters upgraded from earlier versions - do existing namespaces get this label as well).

Reply:
yes, they do :) The correct statement here is that all namespaces (even the old one) get this new label, and it's rewritten.

Actually the label is not persisted as far as I can remember by my tests, but it's still mutated from the original request by the APIServer.

Discussion here: https://github.com/kubernetes/kubernetes/pull/96968#issuecomment-791611582

Reply:
Just to make clear here, I would change this to 
"the Kubernetes API server defaults all namespaces to have this label set to the name of the namespace, even existing namespaces"

Reply:
It sounds like it's a bit stronger than “defaults”, it's enforced and unchangeable. I'll write that - but I'm keen to get this reviewed for technical accuracy!